### ISSUE TITLE
Define screen width when using `capture_screenshot/2`

### ISSUE THREAD

**Issue by RodolfoSilva** [bitcrowd/chromic_pdf#319]
Hi guys,

I need to know if it is possible and how I can set the page size to take a screenshot. Right now, `capture_screenshot/2` does not have an example of how we can define the screen widget.

**Comment by RodolfoSilva** [comment]
Right now, the workaround I've found is to set a global window size with:

```ex
chrome_args: "--window-size=375,375"
```

I was wondering if there is a better way to achieve this result without setting a global value.
@maltoe do you have any suggestion?

**Comment by maltoe** [comment]
Hi @RodolfoSilva,

no, there's no option to set the screen size per job. The `full_page` option comes close, setting the screen size to the content size, see [here](https://github.com/bitcrowd/chromic_pdf/blob/2f1d923c92902c197f94e7775667ae0de7782049/lib/chromic_pdf/pdf/protocols/capture_screenshot.ex#L11-L26). If you'd like to give it a try, you could start from there. Option could be something like `{:device_metrics_override, [width: ..., height: ...]}`.

**Comment by maltoe** [comment]
@RodolfoSilva I've pushed a PR that allows to more easily run custom protocols - something I've been wanting to do for a while. A custom protocol would allow you to execute whatever [devtools](https://chromedevtools.github.io/devtools-protocol/tot/) commands before the actual PDF printing. I've included an example for your use-case in the tests :point_up:. As stated in the PR description, I'd consider the `Protocol` behaviour & the macros still to be internal API, but it's unlikely they're going to change anytime soon.

Still open for a `:device_metrics_override` option if you're willing to work on that instead, though.
